instrumentation/spring/spring-webmvc-3.1/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/springwebmvc/SpringMvcInstrumentationModule.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.springwebmvc;
+
+import static io.opentelemetry.javaagent.tooling.ClassLoaderMatcher.hasClassesNamed;
+
+import com.google.auto.service.AutoService;
+import io.opentelemetry.javaagent.tooling.InstrumentationModule;
+import io.opentelemetry.javaagent.tooling.TypeInstrumentation;
+import java.util.Arrays;
+import java.util.List;
+import net.bytebuddy.matcher.ElementMatcher;
+
+@AutoService(InstrumentationModule.class)
+public final class SpringMvcInstrumentationModule extends InstrumentationModule {
+  public SpringMvcInstrumentationModule() {
+    super("spring-mvc");
+  }
+
+  @Override
+  public ElementMatcher<ClassLoader> classLoaderMatcher() {
+    // Optimization for expensive typeMatchers.
+    return hasClassesNamed(
+        "org.springframework.context.support.AbstractApplicationContext",
```

Comment:
The only unfortunate change is that these optimizations for expensive type matcher aren't next to the optimized type matcher anymore. I don't know if that means it's good to accept classloaderMatcher on instrumentation as well or not, wonder what others think.

Reply:
👍 that would be nice to keep them next to each other somehow

Reply:
Hmm, that is true. I think that we have 2 pattern of `classLoaderMatcher()` usage (correct me if I'm wrong):
1. Optimization for `extends()`/`implements()` `typeMatcher()`
2. library version check, e.g. requiring a class that is present in a particular instrumented library version.

I believe that 2. should stay in the `InstrumentationModule` - 1. should be placed in `TypeInstrumentation` though, as it 's and optimization for a single type instrumentation. I'll add another `classLoaderMatcher()` method to `TypeInstrumentation` and just `and()` those two.